### PR TITLE
Bugfix + sanity checks for stacki_host

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
Started as fixing the documentation for sanity checks, but ended up with a small bugfix (and minor refactors).

##### ISSUE TYPE
- Bugfix Pull Request
- Docs Pull Request

##### COMPONENT NAME
plugins/modules/remote_management/stacki/stacki_host.py

### CODE REVIEW

File: plugins/modules/remote_management/stacki/stacki_host.py [hunk 1]
```diff
@@ -12,53 +12,79 @@
 module: stacki_host
 short_description: Add or remove host to stacki front-end
 description:
- - Use this module to add or remove hosts to a stacki front-end via API.
- - U(https://github.com/StackIQ/stacki)
+  - Use this module to add or remove hosts to a stacki front-end via API.
+  - U(https://github.com/StackIQ/stacki)
```
Comment (felixfontein) [inline plugins/modules/remote_management/stacki/stacki_host.py:16]:
```suggestion
  - Information on stacki can be found at U(https://github.com/StackIQ/stacki).
```

File: plugins/modules/remote_management/stacki/stacki_host.py [hunk 2]
```diff
@@ -12,53 +12,79 @@
...
   prim_intf:
     description:
-     - Name of the primary network interface.
+      - Name of the primary network interface.
+      - Currently not used by the module.
     type: str
   force_install:
     description:
-     - Set value to True to force node into install state if it already exists in stacki.
+      - Set value to True to force node into install state if it already exists in stacki.
```
Comment (felixfontein) [inline plugins/modules/remote_management/stacki/stacki_host.py:56]:
```suggestion
      - Set value to C(true) to force node into install state if it already exists in stacki.
```

File: plugins/modules/remote_management/stacki/stacki_host.py [hunk 3]
```diff
@@ -163,33 +188,25 @@ def __init__(self, module):
...
 
     def stack_check_host(self):
-        res = self.do_request(self.module, self.endpoint, payload=json.dumps({"cmd": "list host"}), headers=self.header, method="POST")
-
-        if self.hostname in res.read():
-            return True
-        else:
-            return False
+        res = self.do_request(self.endpoint, payload=json.dumps({"cmd": "list host"}), headers=self.header, method="POST")
+        return self.hostname in res.read()
```
Comment (felixfontein) [inline plugins/modules/remote_management/stacki/stacki_host.py:201]: Sounds like a bug waiting to happen (`myhost` is assumed to be present when `myhostname` exists). But that's nothing we can easily solve, I guess, without knowing what the API actually returns...

File: plugins/modules/remote_management/stacki/stacki_host.py [hunk 4]
```diff
@@ -258,8 +275,7 @@ def main():
             .format(module.params['name'])
     # Otherwise, state is present, but host doesn't exists, require more params to add host
     elif module.params['state'] == 'present' and not host_exists:
-        for param in ['appliance', 'prim_intf',
-                      'prim_intf_ip', 'network', 'prim_intf_mac']:
+        for param in ['appliance', 'rack', 'rank']:
```
Comment (felixfontein) [inline plugins/modules/remote_management/stacki/stacki_host.py:278]: I would keep the two options in here, and treat them being ignored as a bug that should be fixed (not by this PR :) ).

File: plugins/modules/remote_management/stacki/stacki_host.py (hunk 3 above)
Comment (russoz) [reply plugins/modules/remote_management/stacki/stacki_host.py:201]: Yeah, this module needs a lot of TLC, but yes, a deeper knowledge of the API would be required - I am not planning on that.

**Review by felixfontein — COMMENTED** [review]
Looks good as far as I understand this, looks good.

**Comment by felixfontein** [comment]
@bbyhuy @bsanders as maintainers of this module, can you please take a look?

**Comment by felixfontein** [comment]
If nobody complains, I'll merge by Monday.

**Comment by felixfontein** [comment]
@russoz thanks a lot for fixing this!

**Pull request merged**